### DISCUSSION TITLE
How can i create a Modern Site based on Site Template?

### DISCUSSION THREAD

**Discussion by orbcom-tiagosantos** [pnp/pnpcore#1446] [Q&A]
Hi,

I have application that uses Powershell to apply site template to a new site and its working fine, but now i'm trying to migrate all the workflow to PnP.Core SDK and I'm having issues with this step, "apply site template on a new site".

Can someone help with is problem i have?

Thanks

**Comment by jansenbe** [comment]
@TiagoSantosOrbcom : for now that feature is not in PnP Core SDK, it's however available in PnP Framework (PnP PS also uses these bits from PnP Framework). See here for interop between PnP Core SDK and PnP Framework: https://pnp.github.io/pnpcore/using-the-sdk/basics-pnpframework.html

**Reply by orbcom-tiagosantos** [reply]
@jansenbe thank you for the fast reply :)

Can you give an estimate for applying templates in PnP.Core?
If there is no date set for this, can you put it on the roadmap for implementation?

**Reply by jansenbe** [reply]
@TiagoSantosOrbcom: We've no plans for porting the provisioning engine from PnP Framework to PnP Core. That would be huge work item and given this is an open source project we're dependent of folks in the community to work on this.

**Reply by aferrazlr** [reply]
Hello @jansenbe, to set the correct expectations for this project and clarify the ultimate goal of PnP Core, is it intended to be a feature-complete replacement for the PnP Framework? Will it eventually include the provisioning engine from PnP Core, enabling the application of site templates to new sites?

If PnP Core aims to be a "_new clean replacement for PnP Framework_," a public roadmap for the fully-featured PnP Core would be beneficial for the community, even if it's a flexible plan that evolves over time.

**Reply by jansenbe** [reply]
@TiagoSantosOrbcom : eventually, yes that the plan and let me rephrase that in the short we don't have plans for porting the provisioning engine.

**Reply by aferrazlr** [reply]
@jansenbe, thank you for the clarification. 

Concerning community expectations, are there any plans to publish a roadmap for feature completion? Also, is there an estimated timeline for this plan, such as 1, 3, 5-years?
